general/g.mapsets/list.c

```diff
@@ -35,21 +27,92 @@ void list_accessible_mapsets(const char *fs)
     const char *name;
 
     G_message(_("Accessible mapsets:"));
+
     for (n = 0; (name = G_get_mapset_name(n)); n++) {
         /* match each mapset to its numeric equivalent */
         fprintf(stdout, "%s", name);
         if (G_get_mapset_name(n + 1)) {
-            if (strcmp(fs, "newline") == 0)
-                fprintf(stdout, "\n");
-            else if (strcmp(fs, "space") == 0)
-                fprintf(stdout, " ");
-            else if (strcmp(fs, "comma") == 0)
-                fprintf(stdout, ",");
-            else if (strcmp(fs, "tab") == 0)
-                fprintf(stdout, "\t");
-            else
-                fprintf(stdout, "%s", fs);
+            fprintf(stdout, "%s", fs);
         }
     }
     fprintf(stdout, "\n");
 }
+
+void list_accessible_mapsets_json()
+{
+    int n;
+    const char *name;
+    char *serialized_string = NULL;
+    JSON_Value *root_value = NULL;
+    JSON_Object *root_object = NULL;
+    JSON_Array *mapsets = NULL;
+
+    root_value = json_value_init_object();
+    root_object = json_value_get_object(root_value);
+
+    // Create and add mapsets array to root object
+    json_object_set_value(root_object, "mapsets", json_value_init_array());
+    mapsets = json_object_get_array(root_object, "mapsets");
+
+    // Check that memory was allocated to root json object
+    if (root_value == NULL) {
+        G_fatal_error(_("Failed to initialize JSON. Out of memory?"));
+    }
+
+    // Check that memory was allocated to mapsets array
+    if (mapsets == NULL) {
+        G_fatal_error(_("Failed to initialize JSON array. Out of memory?"));
+    }
+
+    // Add mapsets to mapsets array
+    for (n = 0; (name = G_get_mapset_name(n)); n++) {
```

Comment:
`n` is only (and first) used here, no need to put this at the head of the function.
(You could even declare `name` just before the `for` statement). The same goes for the next function.
```suggestion
    for (int n = 0; (name = G_get_mapset_name(n)); n++) {
```